Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -882,6 +861,54 @@ extension BrowserViewController {
     func hideReaderModeBar(#animated: Bool) {
         readerModeBar.hidden = true
     }
+
+    func enableReaderMode() {
+        if let webView = tabManager.selectedTab?.webView {
+            let backList = webView.backForwardList.backList as [WKBackForwardListItem]
+            let forwardList = webView.backForwardList.forwardList as [WKBackForwardListItem]
+
+            if let currentURL = webView.backForwardList.currentItem?.URL {
+                if let readerModeURL = ReaderModeUtils.encodeURL(currentURL) {
+                    if backList.count > 1 && backList.last?.URL == readerModeURL{
```

Comment:
Nit: Space before `{`

It took me a few seconds to figure out what was going on here, so I'd add a couple comments describing what these are doing (reusing existing history entries to avoid cluttering the backstack with reader URLs).